tests/integration/helpers.py

```diff
@@ -0,0 +1,140 @@
+
+    time = 2
+    for x in range(0, 10):
+        try:
+            _exec_command(instance, ["echo", "'Hello!'"])
+            err = None
+        except Exception as e:
+            err = e
+
+        if err:
+            _logger.info(f"Sleeping for {time} seconds")
+            sleep(time)
+            time *= 2
+        else:
+            break
+
+    if err:
+        raise err
+
+    _exec_commands(
+        instance,
+        [
+            ["ln", "-s", "/bin/true", "/usr/local/bin/udevadm"],
+            ["apt-get", "-y", "update"],
+            ["apt-get", "-y", "install", "ceph-common"],
+            ["snap", "install", "microceph"],
+            ["microceph", "cluster", "bootstrap"],
+            ["microceph", "disk", "add", "loop,2G,3"],
+            ["microceph.ceph", "osd", "pool", "create", f"{FS_NAME}_data"],
+            ["microceph.ceph", "osd", "pool", "create", f"{FS_NAME}_metadata"],
+            ["microceph.ceph", "fs", "new", FS_NAME, f"{FS_NAME}_metadata", f"{FS_NAME}_data"],
+            ["microceph.ceph", "fs", "authorize", FS_NAME, f"client.{USERNAME}", PATH, "rw"],
+            # Need to generate the test files inside microceph itself.
+            [
+                "ln",
+                "-sf",
+                "/var/snap/microceph/current/conf/ceph.client.admin.keyring",
+                "/etc/ceph/ceph.client.admin.keyring",
+            ],
+            [
+                "ln",
+                "-sf",
+                "/var/snap/microceph/current/conf/ceph.keyring",
+                "/etc/ceph/ceph.keyring",
+            ],
+            ["ln", "-sf", "/var/snap/microceph/current/conf/ceph.conf", "/etc/ceph/ceph.conf"],
+        ],
+    )
+
+    # Wait until the cluster is ready to mount the filesystem.
+    time = 2
+    for x in range(0, 10):
+        status = json.loads(_exec_command(instance, ["microceph.ceph", "-s", "-f", "json"]))
+        if status["health"]["status"] == "HEALTH_OK":
+            break
+        _logger.info(f"Sleeping for {time} seconds")
+        sleep(time)
+        time *= 2
+
+    _exec_command(instance, ["mount", "-t", "ceph", f"admin@.{FS_NAME}={PATH}", "/mnt"])
+
+    for i in [1, 2, 3]:
+        _exec_command(instance, ["touch", f"/mnt/test-{i}"])
+
+    return _get_cephfs_info(instance)
+
+
+def _get_cephfs_info(instance: Any) -> Tuple[CephFSShareInfo, CephFSAuthInfo]:
+    status = json.loads(_exec_command(instance, ["microceph.ceph", "-s", "-f", "json"]))
+    fsid = status["fsid"]
+    host = instance.state().network["enp5s0"]["addresses"][0]["address"] + ":6789"
+    key = _exec_command(instance, ["microceph.ceph", "auth", "print-key", f"client.{USERNAME}"])
+    return (
+        CephFSShareInfo(fsid=fsid, name=FS_NAME, path=PATH, monitor_hosts=[host]),
+        CephFSAuthInfo(username=USERNAME, key=key),
+    )
+
+
+def _exec_command(instance: Any, cmd: [str]) -> str:
+    _logger.info(f"Executing `{' '.join(cmd)}`")
+    (code, stdout, stderr) = instance.execute(
```

Comment:
[nit]: You shouldn't need the the parentheses here. Python can unpack properly unpack the return value:

 ```suggestion
    code, stdout, stderr = instance.execute(
```